src/Dscp/Resource/Keys.hs

```diff
@@ -0,0 +1,84 @@
+-- Later this file will also present code related to encrypted keys
+-- management. It will also provide possibility to choose a key from
+-- the predefined test genesis instead of providing a key explicitly.
+
+-- | Key management resource.
+
+module Dscp.Resource.Keys
+    ( KeyParams (..)
+    , KeyResources (..)
+    , krSecretKey
+    , krPublicKey
+    , ourSecretKey
+    , ourPublicKey
+    ) where
+
+import qualified Codec.Serialise as S
+import Control.Exception (Exception)
+import Control.Lens (makeLenses)
+import qualified Data.ByteString.Lazy as BSL
+import Fmt ((+||), (||+))
+import Loot.Base.HasLens (HasLens', lensOf)
+import Loot.Log (logDebug, logError, logInfo)
+import System.Directory as D
+import qualified System.FilePath as FP
+
+import Dscp.Crypto (PublicKey, SecretKey, keyGen, toPublic)
+import Dscp.Resource.Class (AllocResource (..), buildComponentR)
+
+-- Password also (later).
+-- | Parameters needed for key initialisation.
+data KeyParams = KeyParams
+    { kpKeyPath :: FilePath
+      -- ^ Path to the key file.
+    , kpGenKey  :: Bool
+      -- ^ If key on the 'kpKeyPath' is not present, generate it and write it.
+    } deriving (Eq,Ord,Show)
+
+-- In future this will contain encrypted key (maybe).
+-- | Resources related to key management.
+data KeyResources = KeyResources
```

Comment:
@Martoon-00 already did all this for Educator's secret key, and this included on-disk encryption, JSON serialisation and passing a password as a parameter. I really think that it should be generalized somehow in scope of this PR and used for witness keyfile too, because here you are repeating a lot of logic and that's ugly.